src/assignments/yunfeng/lesson_2_answers.clj

```diff
@@ -1,3 +1,23 @@
 (ns lesson-2-answers)
 
 ;;; Enter your answers in here.
+
+(require '[clojure.string :as string])
+
+(defn find-anagrams
+  [vec]
+  (->> (map (fn [str] [str str]) vec)
+       (map (fn [[key value]] {(#(clojure.string/join "" %) (sort (#(string/split % #"") key))) value}))
+       (reduce #(merge-with (fn [v1 v2] (if (set? v1)
+                                          (conj v1 v2)
+                                          #{v1 v2})) %1 %2))
+       (vals)
+       (filter #(set? %))
+       (set)))
+
```

Comment:
Unfortunatly, this contains a subtle bug:  consider:
```
(find-anagrams ["veer" "lake" "item" "kale" "mite" "ever" "everever"])
;; => #{#{"kale" "lake"} #{"item" "mite"} #{"ever" "veer"}}
```
Do you see the problem?

Reply:
I don't know. is it because "everever" should be included in the last set?

Reply:
Your code considers "everever" to be an anagram of "ever".
That's because it considers 2 words to be anagrams of each other
if the set of letters is the same, but this is buggy: i.e.
(set "ever") == (set "everever")
but the two words are not anagrams!  :-(

Reply:
But my code doesn't consider "ever" and "everever" as anagrams of each other. 

and I think my code will consider 2 words are anagrams of each other if the (alphabetically)sorted result of character lists of two strings are the same. For example, the sorted result of "ever" is "eerv" while in the case of "everever" it is "eeeerrvv". So it will not consider these two strings anagram of each other.

Reply:
You are correct, I misread the output of my test.  Good catch!  I'm mistaken.